File: src/Web/aspnet-core/src/Piggyvault.Application/Piggy/Reports/Dto/CategoryReportListDto.cs
namespace Piggyvault.Piggy.Reports.Dto
{
    public class CategoryReportListDto
    {
        public string AccountName { get; set; }
        public decimal Amount { get; set; }
        public decimal AmountInDefaultCurrency { get; set; }
        public string CategoryIcon { get; set; }
        public string CategoryName { get; set; }
        public string CurrencyCode { get; set; }
    }
}

File: src/Mobile/piggy_flutter/lib/models/category_report_list_dto.dart
class CategoryReportListDto {
  CategoryReportListDto({
    this.accountName,
    this.amountInDefaultCurrency,
    this.amount,
    this.categoryIcon,
    this.categoryName,
    this.currencyCode,
  });

  CategoryReportListDto.fromJson(Map<String, dynamic> json)
      : accountName = json['accountName'],
        amount = json['amount'],
        amountInDefaultCurrency = json['amountInDefaultCurrency'],
        categoryIcon = json['categoryIcon'],
        categoryName = json['categoryName'],
        currencyCode = json['currencyCode'];

  final String? accountName, categoryIcon, categoryName, currencyCode;
  final double? amount, amountInDefaultCurrency;
}
